infra/dataform-export/firestore.js

```diff
@@ -0,0 +1,78 @@
+const { Firestore } = require('@google-cloud/firestore')
+
+class FirestoreBatch {
+  constructor (databaseId) {
+    this.db = new Firestore()
+    this.db.settings({ databaseId })
+    this.batchSize = 500
+  }
+
+  async delete () {
+    const collectionRef = this.db.collection(this.collectionName)
+
+    if (this.collectionType === 'report') {
+      console.log('Deleting documents from ' + this.collectionName + ' for date ' + this.date)
+      while (true) {
+        // Query to fetch monthly documents and run delete operations in parallel batches
+        const snapshot = await collectionRef.where('date', '==', this.date).limit(100000).get()
+        if (snapshot.empty) {
+          break
+        }
+
+        const chunks = []
+        for (let i = 0; i < snapshot.docs.length; i += this.batchSize) {
+          chunks.push(snapshot.docs.slice(i, i + this.batchSize))
+        }
+
+        await Promise.all(
+          chunks.map(async (chunk) => {
+            const batch = this.db.batch()
+            chunk.forEach(doc => batch.delete(doc.ref))
+            await batch.commit()
+          })
+        )
```

Comment:
Rewriting the cwv tech report takes around 1h.
Firestore docs limit per batch commit = 500. I checked sequential commits can remove 22K during 1 min.
`adoption` report has 2M rows per month so just deleting it takes 90min.

I've changed it to run 200 commits in parallel (100000 / 500) with `await Promise.all`. 
I haven't seen any API limits per min, so hopefully it scales well for the other reports.